rosetta-runtime/src/main/java/com/rosetta/model/lib/validation/ValidationResult.java

```diff
@@ -1,272 +1,44 @@
 package com.rosetta.model.lib.validation;
 
-import java.util.Optional;
-import java.util.List;
-import java.util.stream.Collectors;
-
 import com.rosetta.model.lib.path.RosettaPath;
 
-import java.util.function.Function;
-
-import static com.rosetta.model.lib.validation.ValidationResult.ValidationType.CHOICE_RULE;
-
-public interface ValidationResult<T> {
-
-	boolean isSuccess();
-
-	String getModelObjectName();
-
-	String getName();
-	
-	ValidationType getValidationType();
+import java.util.Optional;
 
-	String getDefinition();
-	
-	Optional<String> getFailureReason();
-	
-	RosettaPath getPath();
+public class ValidationResult{
+	private static boolean success;
+	private Optional<String> failureReason;
+	private RosettaPath path;
+	private Optional<ValidationData> data;
+
+	@SuppressWarnings("static-access")
+	public ValidationResult(boolean success, String failureReason, RosettaPath path, ValidationData data) {
+		this.success = success;
+		this.failureReason = failureReason!=""?Optional.ofNullable(failureReason):Optional.empty();
+		this.path = path;
+		this.data = data!=null?Optional.ofNullable(data):Optional.empty();
+	}
 
-	static <T> ValidationResult<T> success(String name, ValidationType validationType, String modelObjectName, RosettaPath path, String definition) {
-		return new ModelValidationResult<>(name, validationType, modelObjectName, path, definition, Optional.empty());
+	public static ValidationResult success(RosettaPath path) {
```

Comment:
I have been thinking a bit about backwards compatibility.. I fear that this class is gonna cause problems.

Old generated code will still contain code of the form
```
ValidationResult<Foo> result = ValidationResult.success("foo", ValidationType.CARDINALITY, "Foo", path, "definition");
```
which will suddenly stop working because of multiple reasons:
- The type parameter `<Foo>` is now invalid, since we got rid of the generic `T` type.
- The method `ValidationResult.success(String, ValidationType, String, RosettaPath, String)` does not exist anymore.
- The type `ValidationResult.ValidationType` does not exist anymore.

I'm not sure yet about what our best course of action is.